### PR TITLE
feat: add `iter/cusome`

### LINKED ISSUE #2334: [RFC]: add `@stdlib/iter/cusome`
### Description

This RFC proposes adding the package `@stdlib/iter/cusome`, which cumulatively tests whether at least `n` iterated values are truthy. The function should be a transform iterator, continuing to iterate while source iterator values are available.

```javascript
var array2iterator = require( '@stdlib/array/to-iterator' );

var arr = array2iterator( [ false, false, false, true, true ] );

var it = iterCuSome( arr, 2 );

var v = it.next().value;
// returns false

v = it.next().value;
// returns false

v = it.next().value;
// returns false

v = it.next().value;
// returns false

v = it.next().value;
// returns true

var bool = it.next().done;
// returns true
```

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/iter/some`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by TheNourhan**
Resolves #2334

## Description

> What is the purpose of this pull request?

This pull request:

- Implementation of the iterCuSome function in the lib folder.
- Documentation updates in the docs folder:
    - Added REPL usage examples to docs/repl.txt.
    - Created TypeScript type definitions in the docs/types folder.
- Updated README.md.
- Added benchmarks to the benchmark folder.
- Added unit tests to the test folder.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2334

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@TheNourhan I think you misunderstood the RFC. The iterator should never end until the source iterator ends. It should cumulatively test for at least n elements, forever returning true until the source iterator no longer has iterated values. See other cumulative iterators for an example.

Also, please ensure that your source implementation matches the style and conventions of other iterator packages.

**Comment by TheNourhan** [comment]
@kgryte Sorry for the confusion, I've edited it

File: lib/node_modules/@stdlib/iter/cusome/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		throw new TypeError( format( 'invalid argument. First argument must be an iterator. Value: `%s`.', iterator ) );
+	}
+	if ( !isPositiveInteger( n ) ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be a positive integer. Value: `%s`.', n ) );
+	}
+
+	// Create an iterator protocol-compliant object:
+	iter = {};
+	count = 0;
+	setReadOnly(iter, 'next', next);
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cusome/lib/main.js]: This iterator is missing a `return` method and an iterator symbol check. See https://github.com/stdlib-js/stdlib/blob/903c51c7d0a06d9186a6f2be1b01fa25f770a3eb/lib/node_modules/%40stdlib/iter/cuany/lib/main.js#L76. The implementation should be very similar to what is found in `cuany`.

File: lib/node_modules/@stdlib/iter/cusome/test/test.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		v = iterator.next();
+		if ( v.done ) {
+			break;
+		}
+		actual.push( v.value );
+	}
+
+	t.deepEqual( actual, expected, 'returns expected values' );
+	t.end();
+});
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cusome/test/test.js:343]: As in `cuany`, you need iterator symbol tests.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for continuing to work on this, @TheNourhan. This is coming along. Left a couple more comments. I suggest more closely studying `cuany` and updating this PR accordingly.

**Comment by kgryte** [comment]
Additionally, CI failures also need to be addressed before this PR can be merged.

**Comment by TheNourhan** [comment]
@kgryte I made all the changes you requested, I hope everything is fine now :)

**Comment by TheNourhan** [comment]
If there something I can improve to move forward with the merger? @kgryte

**Review by kgryte — APPROVED** [review]
LGTM. Thanks for working on this, @TheNourhan. I made a number of changes which you'll want to keep in mind for future PRs. The main thing is being consistent with code elsewhere in the project.

**Review by kgryte — APPROVED** [review]

**Pull request merged**